connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java

```diff
@@ -55,18 +60,51 @@ public void sendUserNotification(UserNotification userNotification, Acknowledgme
         log.debug(LogUtils.CONFIDENTIAL_MARKER, "send user notification = {}", userNotification);
         if (validateProductTopic(userNotification.getProductId())) {
             NotificationToSend notificationToSend = notificationMapper.createUserNotification(userNotification);
-            notificationToSend.setType(NotificationType.getNotificationTypeFromRelationshipState(userNotification.getUser().getRelationshipStatus()));
-            String userNotificationToSend = mapper.writeValueAsString(notificationToSend);
-            String topic = producerAllowedTopics.get().get(userNotification.getProductId());
-            String logSuccess = String.format("sent notification for user : %s, to FD", userNotification.getUser().getUserId());
-            String logFailure = String.format("error during notification sending for user %s: {}, on FD ", userNotification.getUser().getUserId());
-            sendNotification(userNotificationToSend, topic, logSuccess, logFailure, acknowledgment);
+            if (userNotification.getEventType().equals(QueueEvent.UPDATE) && userNotification.getUser().getRelationshipStatus() == null) {
+                UserProductDetails userProduct = externalApiConnector.getUserOnboardedProductDetails(userNotification.getUser().getUserId(), userNotification.getInstitutionId(), userNotification.getProductId());
+                if(userProduct.getOnboardedProductDetails() != null) {
+                    sendUpdateUserEvents(notificationToSend, userProduct);
+                    acknowledgment.acknowledge();
+                }
+            } else {
+                notificationToSend.setType(NotificationType.getNotificationTypeFromRelationshipState(userNotification.getUser().getRelationshipStatus()));
+                String userNotificationToSend = mapper.writeValueAsString(notificationToSend);
+                String topic = producerAllowedTopics.get().get(userNotification.getProductId());
+                String logSuccess = String.format("sent notification for user : %s, to FD", userNotification.getUser().getUserId());
+                String logFailure = String.format("error during notification sending for user %s: {}, on FD ", userNotification.getUser().getUserId());
```

Comment:
these lines are duplicated....you could move them to sendNotification method

Reply:
the log message for FD and SAP have different formats, I will indroduce a CONSTANT for them